lib/grp.gi

```diff
@@ -130,14 +130,103 @@ end);
 InstallOtherMethod(MinimalGeneratingSet,"fallback method to inform user",true,
   [IsObject],0,
 function(G)
+  local
+    check,GbyGk,
+    Gkm1byGk,
+    Gkm1byGk_L,
+    Gkm1byGk_elem_reps,
+    Gkm1byGk_gen,
+    Gkm1byGk_gen_reps,
+    mingenset_km1_reps,
+    mingenset_k_reps,
+    temp,i,j,l,L,x,xl,y,prev,gmod,N,g,g0,g1,s,r,stop,
+    cs,phi_GbyG1,GbyG1,phi_GbyGk,phi_Gkm1byGk,k;
   if IsGroup(G) and IsSolvableGroup(G) then
     TryNextMethod();
+  elif IsGroup(G) and IsFinite(G) then
+    cs := ChiefSeries(G);
+    phi_GbyG1 := NaturalHomomorphismByNormalSubgroup(G,cs[2]);
+    GbyG1 := ImagesSource(phi_GbyG1);
+    mingenset_k_reps := List(MinimalGeneratingSet(GbyG1), x -> PreImagesRepresentative(phi_GbyG1, x));
+    # GbyG1 is a simple group, so it can be handelled by GbyG1
+    for k in [3..Length(cs)] do # Lifting
+      mingenset_km1_reps := mingenset_k_reps;
```

Comment:
It seems `mingenset_km1_reps` is redundant, and all occurrences can be replaced by `mingenset_k_reps`?

Reply:
I kept it there since I thought it would help in understanding the algorithm. But yes, it's redundant.